dojo/tools/api_msdefender/api_client.py

```diff
@@ -0,0 +1,55 @@
+import json
+import urllib.request
+import urllib.parse
+import requests
+from dojo.utils import prepare_for_view
+
+
+class MSDefenderAPI:
+    """
+    A simple client for the MS Defender Vulnerability API
+    """
+
+    def __init__(self, tool_config):
+        if tool_config.authentication_type == "Password":
+            self.base_url = "https://login.microsoftonline.com/%s/oauth2/token" % (tool_config.extras)
+            body = {
+                'resource': 'https://api.securitycenter.microsoft.com',
+                'client_id': tool_config.username,
+                'client_secret': prepare_for_view(tool_config.password),
+                'grant_type': 'client_credentials'
+            }
+        else:
+            raise ValueError(
+                "Authentication type {} not supported".format(
+                    tool_config.authentication_type
+                )
+            )
+        data = urllib.parse.urlencode(body).encode("utf-8")
+        req = urllib.request.Request(self.base_url, data)
+        response = urllib.request.urlopen(req)
+        jsonResponse = json.loads(response.read())
+        self.aadToken = jsonResponse["access_token"]
+
+    def get_findings(self):
+        page = 0
+        results = []
+        while True:
+            endpoint = 'https://api.securitycenter.microsoft.com/api/vulnerabilities/machinesVulnerabilities?$skip=' + str(page)
+            headers = {"Authorization": 'Bearer ' + self.aadToken}
+            json_output = requests.get(endpoint, headers=headers).json()
+            if "error" in json_output:
+                if json_output["error"]["code"] == "TooManyRequests":
+                    import time
+                    time.sleep(60)  # If you make too many requests per minute, the API lets you wait for 60 seconds.
```

Comment:
- This may be problematic. Not sure what the timeout is for `api/v2/import` but the whole process might drop.
- Plus, if you receive `TooManyRequests`, you are not performing retry but following on the next page
- Using of 
   ```python
    try:
         json_output.raise_for_status()
    except requests.exceptions.HTTPError:
        if json_output.status_code == 429:  # HTTP-429 == "TooManyRequests"
            ...
        else:
            raise
   ```
    might be better idea